src/wallet/rpc/wallet.cpp

```diff
@@ -637,68 +655,136 @@ RPCHelpMan simulaterawtransaction()
         filter |= ISMINE_WATCH_ONLY;
     }
-
+    // fetch transactions
     const auto& txs = request.params[0].get_array();
     CAmount changes{0};
-    std::map<COutPoint, CAmount> new_utxos; // UTXO:s that were made available in transaction array
-    std::set<COutPoint> spent;
-
+    // UTXOs that were made available in transaction array
+    std::map<COutPoint, CAmount> new_utxos;
+    // Tracks the indices of the txs spending an outpoint
+    std::map<COutPoint, std::set<int>> outpoint_to_tx;
+    // Tracks conflicting transactions(provided by the user) given a txid
+    std::map<uint256, std::set<int>> internal_conflicts;
+    // Tracks set of txids in the wallet that are conflicting with the provided transaction given txid of tx provided by the user
+    std::map<uint256, std::set<std::string>> txids;
+    // indices of duplicate transactions
+    std::set<int> duplicates;
+    //Tracks txid given index for the tx provided by the user
+    std::map<int, uint256> txhash;
+    // used for dealing with duplicates
+    std::map<uint256, bool> done;
     for (size_t i = 0; i < txs.size(); ++i) {
         CMutableTransaction mtx;
+        // decoding ith transaction hex
         if (!DecodeHexTx(mtx, txs[i].get_str(), /* try_no_witness */ true, /* try_witness */ true)) {
             throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "Transaction hex string decoding failure.");
         }
-
+        const auto hash = mtx.GetHash();
+        done[hash] = false;
+        // if the tx is a duplicate one then just add it to duplicates
+        if (txids.count(hash)) {
+            duplicates.insert(i);
+            txhash[i] = hash;
+            continue;
+        }
         // Fetch previous transactions (inputs)
         std::map<COutPoint, Coin> coins;
         for (const CTxIn& txin : mtx.vin) {
             coins[txin.prevout]; // Create empty map entry keyed by prevout.
         }
         pwallet->chain().findCoins(coins);
-
         // Fetch debit; we are *spending* these; if the transaction is signed and
         // broadcast, we will lose everything in these
         for (const auto& txin : mtx.vin) {
             auto& op = txin.prevout;
-            if (spent.count(op)) {
-                throw JSONRPCError(RPC_INVALID_PARAMETER, "Transaction(s) are spending the same output more than once");
-            }
-            if (new_utxos.count(op)) {
-                changes -= new_utxos.at(op);
-                new_utxos.erase(op);
-            } else {
-                if (!coins.count(op)) {
-                    throw JSONRPCError(RPC_INVALID_PARAMETER, "One or more transaction inputs are missing");
-                }
-                if (coins.at(op).IsSpent()) {
-                    throw JSONRPCError(RPC_INVALID_PARAMETER, "One or more transaction inputs have been spent already");
+            // if user tries to double spend in the provided transactions themselves then without this condition we will get an inappropirate error
```

Comment:
In bf9deb407a8d73ac3b645b34c9bd6d28729a809c "rpc/wallet: Add details and duplicate section for simulaterawtransaction"

This comment is a little confusing to understand.

``` suggestion
            // Only apply changes the first time a transaction is seen. Otherwise duplicate transactions will give incorrect errors.
```